android/java/org/mozilla/vrb/ImageLoader.java

```diff
@@ -41,6 +44,30 @@ private static void loadFromAssets(final AssetManager aAssets, final String aFil
                     }
                 }
             }
+        } catch(Exception e) {
+            String reason = "Error loading image file: " + aFileName + " (" + e.toString() + ")";
+            ImageLoadFailed(aFileReader, aTrackingHandle, reason);
+            return;
+        }
+    }
+
+    @Keep
+    private static void loadFromAssets(final AssetManager aAssets, final String aFileName, final long aFileReader, final int aTrackingHandle) {
+        try {
+            InputStream in = aAssets.open(aFileName, AssetManager.ACCESS_STREAMING);
+            loadFromInputStream(in, aFileName, aFileReader, aTrackingHandle);
+        } catch(IOException e) {
+            String reason = "Error loading image file: " + aFileName + " (" + e.toString() + ")";
+            ImageLoadFailed(aFileReader, aTrackingHandle, reason);
+            return;
+        }
+    }
+
+    @Keep
+    private static void loadFromRawFile(final String aFileName, final long aFileReader, final int aTrackingHandle) {
+        try {
```

Comment:
Should this be:

try (InputStream in = new FileInputStream) {

(I wouldn't even know about that if you hadn't told me about it. 😅)